broker/lib/fabrik/UnlockResourcePoller.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const pubsub = require('pubsub-js');
+const eventmesh = require('../../../eventmesh');
+const CONST = require('../constants');
+const logger = require('../logger');
+const lockManager = require('./../../../eventmesh').lockManager;
+const errors = require('../errors');
+const NotFound = errors.NotFound;
+
+class UnlockResourcePoller {
+  static start() {
+    function poller(object, interval) {
+      const lockDetails = JSON.parse(object.spec.options);
+      return eventmesh.server.getResource(lockDetails.lockedResourceDetails.resourceType, lockDetails.lockedResourceDetails.resourceName, lockDetails.lockedResourceDetails.resourceId)
+        .then((resource) => {
+          const resourceState = resource.body.status.state;
+          logger.debug(`[Unlock Poller] Got resource ${lockDetails.lockedResourceDetails.resourceId} state as `, resourceState);
+          if (resourceState === CONST.APISERVER.RESOURCE_STATE.SUCCEEDED || resourceState === CONST.APISERVER.RESOURCE_STATE.FAILED || resourceState === CONST.APISERVER.RESOURCE_STATE.ERROR) {
+            return lockManager.unlock(object.metadata.name)
+              .then(() => clearInterval(interval));
+          }
+        })
+        .catch(NotFound, err => {
+          logger.info('Resource not found : ', err);
+          return lockManager.unlock(object.metadata.name)
```

Comment:
What happens if unlock fails here?

Reply:
In unlock method we have put 3 retries. Even then if unlock fails poller will keep retrying to unlock